constraint/pkg/client/drivers/k8scel/driver_test.go

```diff
@@ -0,0 +1,474 @@
+					{
+						Expression: `object.metadata.name == "unrecognizable-name"`,
+						Message:    "unexpected name",
+					},
+				},
+			}, nil),
+			constraint:         makeConstraint(nil),
+			isAdmissionRequest: true,
+			vapDefault:         ptr.To[vapDefault](VAPDefaultNo),
+			expectedViolations: true,
+		},
+		{
+			name: "Unsatisfied constraint, default assume no VAP, admission request, template override",
+			template: makeTemplateWithSource(&schema.Source{
+				Validations: []schema.Validation{
+					{
+						Expression: `object.metadata.name == "unrecognizable-name"`,
+						Message:    "unexpected name",
+					},
+				},
+			}, ptr.To[string](string(VAPDefaultYes))),
+			constraint:         makeConstraint(nil),
+			isAdmissionRequest: true,
+			vapDefault:         ptr.To[vapDefault](VAPDefaultNo),
+			expectedViolations: false,
+		},
+		{
+			name: "Unsatisfied constraint, default assume no VAP, admission request, constraint override",
+			template: makeTemplateWithSource(&schema.Source{
+				Validations: []schema.Validation{
+					{
+						Expression: `object.metadata.name == "unrecognizable-name"`,
+						Message:    "unexpected name",
+					},
+				},
+			}, nil),
+			constraint:         makeConstraint(ptr.To[string](string(VAPDefaultYes))),
+			isAdmissionRequest: true,
+			vapDefault:         ptr.To[vapDefault](VAPDefaultNo),
+			expectedViolations: true,
+		},
+		{
+			name: "Unsatisfied constraint, default assume VAP, admission request, constraint override",
+			template: makeTemplateWithSource(&schema.Source{
+				Validations: []schema.Validation{
+					{
+						Expression: `object.metadata.name == "unrecognizable-name"`,
+						Message:    "unexpected name",
+					},
+				},
+			}, nil),
+			constraint:         makeConstraint(ptr.To[string](string(VAPDefaultNo))),
+			isAdmissionRequest: true,
+			vapDefault:         ptr.To[vapDefault](VAPDefaultYes),
+			expectedViolations: true,
+		},
+		{
+			name: "Unsatisfied constraint, default assume VAP, admission request, constraint template override",
+			template: makeTemplateWithSource(&schema.Source{
+				Validations: []schema.Validation{
+					{
+						Expression: `object.metadata.name == "unrecognizable-name"`,
+						Message:    "unexpected name",
+					},
+				},
+			}, ptr.To[string](string(VAPDefaultNo))),
+			constraint:         makeConstraint(nil),
+			isAdmissionRequest: true,
+			vapDefault:         ptr.To[vapDefault](VAPDefaultYes),
+			expectedViolations: true,
+		},
+		{
+			name: "Unsatisfied constraint, VAP disabled, all override",
+			template: makeTemplateWithSource(&schema.Source{
+				Validations: []schema.Validation{
+					{
+						Expression: `object.metadata.name == "unrecognizable-name"`,
+						Message:    "unexpected name",
+					},
+				},
+			}, ptr.To[string](string(VAPDefaultYes))),
+			constraint:         makeConstraint(ptr.To[string](string(VAPDefaultYes))),
+			isAdmissionRequest: true,
+			expectedViolations: true,
```

Comment:
VAP generation is presumed disabled because no default is provided to the engine. In the case where VAP generation is presumed disabled, the CF is on the hook for enforcing all policies.

This feature is useful for something like the Gator CLI, or any enforcement point where VAP is not available (basically anywhere but k8s admission).

Reply:
I see. The difference is whether  vapDefault was set or not. In the case it’s set e.g ptr.To[vapDefault](VAPDefaultNo),  it can be overridden but if not set, it cannot.

Reply:
Updated test description to flag that default == nil